scipy/signal/_spectral_py.py

```diff
@@ -267,6 +269,12 @@ def periodogram(x, fs=1.0, window='boxcar', nfft=None, detrend='constant',
 
     if window is None:
         window = 'boxcar'
+    elif hasattr(window, 'size'):
+        axis_size = x.shape[axis]
+        window_size = window.size
+        if window_size != axis_size:
```

Comment:
Almost right, it looks like this check must happen after `nperseg` is determined (because it isn't always equal to `x.shape[axis]` and the correct criterion is if window_size != nperseg`. The error message does look fine.